### PR TITLE
Add meta migration tool

### PR DESCRIPTION
**Pull request by longjiquan**
Signed-off-by: longjiquan <jiquan.long@zilliz.com>
issue: #19708

### CODE REVIEW

File: cmd/tools/migration/migration/runner.go [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+func (r *Runner) Rollback() error {
+	source, err := backend.NewBackend(r.cfg.MilvusConfig, r.cfg.SourceVersion)
+	if err != nil {
+		return err
+	}
+	target, err := backend.NewBackend(r.cfg.MilvusConfig, r.cfg.TargetVersion)
+	if err != nil {
+		return err
+	}
+	if err := source.Clean(); err != nil {
```
Comment (czs007) [inline cmd/tools/migration/migration/runner.go:171]: if source equal target, Clean may only to be executed once ?

File: cmd/tools/migration/utils/util.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		return "", 0, NewErrNotOfTsKey(s)
+	}
+	convertedTs, err := strconv.Atoi(got[1])
+	if err != nil {
+		return "", 0, fmt.Errorf("%s is not of snapshot", s)
+	}
+	return got[0], Timestamp(convertedTs), nil
+}
+
+func GetFileName(path string) string {
```
Comment (czs007) [inline cmd/tools/migration/utils/util.go:45]: Golang filePath package is what you need. just for remind.

File: cmd/tools/migration/versions/version.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	Version220, _ = semver.Parse(version220Str)
+	VersionMax, _ = semver.Parse(VersionMaxStr)
+}
+
+func Range21x(version semver.Version) bool {
+	return version.GTE(Version210) && version.LT(Version220)
+}
+
+func Range22x(version semver.Version) bool {
+	return version.GTE(Version220)
```
Comment (czs007) [inline cmd/tools/migration/versions/version.go:28]: && vertion.LT(Version230)?

File: cmd/tools/migration/utils/util.go (hunk 2 above)
Comment (longjiquan) [reply cmd/tools/migration/utils/util.go:45]: OK. `path.Base` will replace this.

File: cmd/tools/migration/versions/version.go (hunk 3 above)
Comment (longjiquan) [reply cmd/tools/migration/versions/version.go:28]: OK.

**Pull request merged**